subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -293,3 +310,132 @@ export function handleGovSetReserveFee(event: GovSetReserveFee): void {
     event,
   );
 }
+
+export function handleStake(event: Stake): void {
+  const underlying = event.params.underlying.toHexString();
+  // find market id for underlying
+  const assetStatus = getOrCreateAssetStatus(underlying);
+  const marketId = assetStatus.eToken!;
+  const market = getOrCreateMarket(marketId);
+  const deltaStakedAmount = getDeltaStakeAmount(event);
+
+  // keep track of staked amount from epoch 1
+  const epochID = getCurrentEpoch(event);
+  if (epochID < 0) {
+    market._stakedAmount = market._stakedAmount.plus(deltaStakedAmount);
+    market.save();
+    return;
+  }
+
+  const epochStartBlock = getStartBlockForEpoch(epochID)!;
+  let epoch = _Epoch.load(epochID.toString());
```

Comment:
each epoch is ~6 minutes, does that mean the emissions are updated every 6 minutes? Or is it usually the same rewards so we are just updating the price?

Reply:
hmmm, we should put that in there. So from my understanding the price only gets updated every 15 days? It could change pretty drastically. We should have logic to update the price at least every 30 mins.

Reply:
Good question. They only ever distribute rewards at the end of each epoch, so the RewardTokenEmissionAmount will not change within an epoch. Do you think it is necessary to update RewardTokenEmissionUSD every 30 minutes? If so, we can update RewardTokenEmissionUSD within `updatePrices()` function when we update prices for all markets (there is a EUL market, so EUL price will be update there).

Reply:
Yes I think so. In all of the other protocol rewards we calculate the rate based on the current price of EUL. That sounds like the right spot ✅

Reply:
OK, added code to update rewardTokenEmissionsUSD when EUL price is updated: https://github.com/messari/subgraphs/pull/1429/commits/939d3fd1b56e1321ec78aff6a77946208cc62404

Reply:
OK, sync'ed successfully. I will merge.